src/univariate/continuous/tdist.jl

```diff
@@ -82,7 +82,7 @@ end
 function rand(rng::AbstractRNG, d::TDist)
     ν = d.ν
     z = sqrt(rand(rng, Chisq{typeof(ν)}(ν)) / ν)
-    return randn(rng) / (isinf(ν) ? one(z) : z)
+    return randn(rng, typeof(ν)) / (isinf(ν) ? one(z) : z)
```

Comment:
IMO it shouldn't depend on the parameter (which could any non-floating point number) but floating point number `z` (whose type we do not want to mess up, similar to also the case of the `Gamma` sampler above):
```suggestion
    return randn(rng, typeof(z)) / (isinf(ν) ? one(z) : z)
```